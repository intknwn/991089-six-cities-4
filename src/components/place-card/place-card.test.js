import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCard from './place-card.jsx';

const place = {
  "bedrooms": 3,
  "city": {
    "location": {
      "latitude": 50.846557,
      "longitude": 4.351697,
      "zoom": 13
    },
    "name": `Brussels`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "goods": [`Wahser`, `Fridge`, `Towels`, `Breakfast`, `Baby seat`, `Dishwasher`, `Laptop friendly workspace`, `Air conditioning`],
  "host": {
    "avatar_url": `img/avatar-angelina.jpg`,
    "id": 25,
    "is_pro": true,
    "name": `Angelina`
  },
  "id": 1,
  "images": [
    `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
  ],
  "is_favorite": false,
  "is_premium": false,
  "location": {
    "latitude": 50.842557,
    "longitude": 4.3536969999999995,
    "zoom": 16
  },
  "max_adults": 1,
  "preview_image": `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
  "price": 227,
  "rating": 2.2,
  "title": `Nice, cozy, warm big bed apartment`,
  "type": `room`
};

it(`PlaceCard render`, () => {
  const tree = renderer.create(
      <PlaceCard
        place={place}
        onActiveItemSet={() => {}}
        onCardTitleClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
