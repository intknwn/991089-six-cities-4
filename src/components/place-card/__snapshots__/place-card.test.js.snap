// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard render 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://via.placeholder.com/260x200?text=Place+1"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          230
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button  button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "20%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Comfortable Double Room near Amsterdam Center
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      room
    </p>
  </div>
</article>
`;
